File: SDKs/amazonaws.com/nimble/2020-08-01/python/src/sdk/models/shared/studiomembership.py
from dataclasses import dataclass, field
from typing import Enum,Optional
from dataclasses_json import dataclass_json
from . import studiopersona_enum


@dataclass_json
@dataclass
class StudioMembership:
    identity_store_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'identityStoreId' }})
    persona: Optional[studiopersona_enum.StudioPersonaEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'persona' }})
    principal_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'principalId' }})
    

File: SDKs/amazonaws.com/nimble/2020-08-01/typescript/src/sdk/models/shared/studiomembership.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { StudioPersonaEnum } from "./studiopersonaenum";



export class StudioMembership extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=identityStoreId" })
  identityStoreId?: string;

  @SpeakeasyMetadata({ data: "json, name=persona" })
  persona?: StudioPersonaEnum;

  @SpeakeasyMetadata({ data: "json, name=principalId" })
  principalId?: string;
}
